File: thrift-php/MongoTestHandler.py
#!/usr/bin/env python
#encoding=utf-8

import sys
sys.path.append('./gen-py')
import pymongo
import json
from mongotest import MongoTest

class MongoTestHandler(MongoTest.Iface):

    def __init__(self):
        #self.con = pymongo.Connection()
        pass

    def getServerStatus(self, ):
        #data = self.con.admin.command(pymongo.son_manipulator.SON([('serverStatus', 1)]))
        #return json.dumps(data['connections'])
        return 'test'

File: thrift-php/src/MongoTestHandler.java

import org.apache.thrift.TException;

import com.alibaba.fastjson.JSON;
import com.mongodb.BasicDBObject;
import com.mongodb.CommandResult;
import com.mongodb.DB;
import com.mongodb.DBObject;
import com.mongodb.Mongo;

import mongotest.*;

public class MongoTestHandler implements mongotest.MongoTest.Iface{

    public String getServerStatus(){

        try{
            Mongo m = new Mongo();
            DB db = m.getDB("admin");
            DBObject cmd = new BasicDBObject();
            cmd.put("serverStatus", 1);
            CommandResult cr = db.command(cmd);
            Object b = cr.get("connections");
            String s = JSON.toJSONString(b);
            return s;
        }catch(Exception e){
            return "Exception";
        }
    }

}
